test/integration/mergeLots.js

```diff
@@ -0,0 +1,286 @@
+              'Verify lot3 no longer exists');
+          });
+      })
+      .then(() => {
+        db.exec(`SELECT HEX(lot_uuid) as lot_uuid from lot_tag WHERE tag_uuid = 0x${tag3Uuid}`)
+          .then((res) => {
+            expect(res.length).to.equal(1);
+            expect(res[0].lot_uuid).to.equal(lot1Uuid,
+              'Verify that tag3 now refers to lot1');
+          });
+      })
+      .then(() => {
+        db.exec(`SELECT HEX(lot_uuid) as lot_uuid from stock_movement WHERE uuid = 0x${stockMovement1Uuid}`)
+          .then((res) => {
+            expect(res.length).to.equal(1);
+            expect(res[0].lot_uuid).to.equal(lot1Uuid,
+              'Verify that the stock movement now refers to lot1');
+          });
+      });
+  });
+
+  // ---------------------------------------------------------------------------
+
+  it('Merge lots 4 and 5 with lot 1 (multiple lots)', () => {
+    return agent.post(`/lots/${lot1Uuid}/merge`)
+      .send({ lotsToMerge : [lot4Uuid, lot5Uuid] })
+      .then(res => {
+        // Verify the operation was successful
+        expect(res).to.have.status(200);
+      })
+      .then(() => {
+        db.exec(`SELECT * from lot WHERE uuid IN (0x${lot4Uuid}, 0x${lot5Uuid})`)
+          .then((res) => {
+            expect(res.length).to.equal(0,
+              'Verify lots 4 and 5 no longer exist');
+          });
+      })
+      .then(() => {
+        db.exec(`SELECT HEX(lot_uuid) as lot_uuid from lot_tag WHERE tag_uuid = 0x${tag4Uuid}`)
+          .then((res) => {
+            expect(res.length).to.equal(1);
+            expect(res[0].lot_uuid).to.equal(lot1Uuid,
+              'Verify that tag4 now points to lot1');
+          });
+      })
+      .then(() => {
+        db.exec(`SELECT HEX(lot_uuid) as lot_uuid from lot_tag WHERE tag_uuid = 0x${tag5Uuid}`)
+          .then((res) => {
+            expect(res.length).to.equal(1);
+            expect(res[0].lot_uuid).to.equal(lot1Uuid,
+              'Verify that tag5 now points to lot1');
+          });
+      });
+  });
+
+  // ===========================================================================
+  // Cleanup
+
+  it(`Deleting all temporary mock lots, tags, lot tags, and stock movements`, () => {});
+
+  after('Delete temporary tags and lot tags', () => {
+    return mockTags.reduce((chain, p) => {
+      return chain
+        .then(() => db.exec(`DELETE FROM lot_tag WHERE lot_uuid=0x${p[1]};`))
+        .then(() => db.exec(`DELETE FROM tags WHERE uuid=0x${p[0]};`));
+    }, Promise.resolve());
+  });
+
+  after('Delete temporary stock movements', () => {
+    return mockStockMovements.reduce((chain, p) => {
+      return chain
+        .then(() => db.exec(`DELETE FROM stock_movement WHERE uuid=0x${p[0]};`));
+    }, Promise.resolve());
+  });
+
+  after('Delete temporary lots', () => {
+    return mockLots.reduce((chain, p) => {
+      return chain
+        .then(() => db.exec(`DELETE FROM lot WHERE uuid=0x${p[0]};`));
```

Comment:
This works fine.  I have no issue with it as written.

However, since it doesn't matter the order in which you delete things, a more readable version might by to compose a single SQL statement using `WHERE uuid IN(?)`.  In fact, `db.exec()` will do this for you automagically.

```js
after('Delete temporary stock movements', () => {
  const lotUuids = mockLots.map(lot => db.bid(lot[0]));
  return db.exec(`DELETE FROM lot WHERE uuid IN (?);`, lotUuids);
});
```

Reply:
Will do this next.

Reply:
Done. Thanks for the suggestion!